From 23fa41ce77279b54104b3cf0be92a6a043348631 Mon Sep 17 00:00:00 2001
From: allen yan <yanwei@marvell.com>
Date: Mon, 13 Feb 2017 18:13:03 +0800
Subject: [PATCH 0832/1345] mtd: spi-nand: add NAND_CMD_STATUS command for
 giga device driver

commit  203fb976860d8858f79e7e604f171dc291e2a7d4 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

- The original giga device driver doesn't support NAND_CMD_STATUS
  command. But the MTD architecture needs NAND_CMD_STATUS to return
  nand status. e.g. in function nand_check_wp, it needs use command
  NAND_CMD_STATUS to return the nand is write protected or not.
- This patch implement NAND_CMD_STATUS command to return the nand
  status.

Change-Id: Iac728b612c7de73f0281bfe4a0e01962bcc96e8b
Signed-off-by: allen yan <yanwei@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/37183
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hua Jing <jinghua@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/staging/gd5f_spinand/gd5f_spinand.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/staging/gd5f_spinand/gd5f_spinand.c b/drivers/staging/gd5f_spinand/gd5f_spinand.c
index eb40c4b..c673485 100644
--- a/drivers/staging/gd5f_spinand/gd5f_spinand.c
+++ b/drivers/staging/gd5f_spinand/gd5f_spinand.c
@@ -26,6 +26,7 @@
 #define MAX_WAIT_JIFFIES		(40 * HZ)
 #define MAX_WAIT_ERASE_JIFFIES		((HZ * 400) / 1000)
 #define AVERAGE_WAIT_JIFFIES		((HZ * 20) / 1000)
+#define STAT_MASK			0x80
 
 #ifdef CONFIG_MTD_SPINAND_ONDIEECC
 
@@ -865,6 +866,14 @@ static void spinand_cmdfunc(struct mtd_info *mtd, unsigned int command,
 				__func__);
 		spinand_reset(info->spi);
 		break;
+	case NAND_CMD_STATUS:
+		spinand_get_protection(info->spi, state->buf);
+		if (!(state->buf[0] & STAT_MASK))
+			state->buf[0] |= STAT_MASK;
+		else
+			state->buf[0] &= ~STAT_MASK;
+		state->buf_ptr = 0;
+		break;
 	default:
 		dev_err(&mtd->dev, "Command 0x%x not implementd or unknown.\n",
 			command);
-- 
1.7.9.5

